Support Mariadb 12.3 InnoDB Binlog (#7711)
debezium/debezium: documentation/modules/ROOT/partials/modules/all-connectors/shared-mariadb-mysql.adoc

Reviewer: ```suggestion
* xref:enabling-the-{context}-binlog-for-debezium[]
* xref:enabling-innodb-binlog-for-debezium[]

```

* xref:creating-a-{context}-user-for-a-debezium-connector[]
* xref:enabling-the-{context}-binlog-for-debezium[]

> * xref:enabling-innodb-binlog-for-debezium[]

Reviewer: ```suggestion
ifdef::community[]
* xref:enabling-innodb-binlog-for-debezium[]
endif::community[]
```
Arrg. Sorry. I meant to also conditionalize this! I really inserted it as a reminder to future me that for the product edition we'd want to expose it here, but of course, for now, it's not productized. :/

end::enabling-binlog[]

ifdef::MARIADB[]

> === Enabling the InnoDB-based binary log (MariaDB 12.3 and later)

Reviewer: ```suggestion
=== Enabling the InnoDB-based binary log (MariaDB 12.3 and later)

tag::enabling-innodb-binlog[]
ifdef::community[]
```
The `MARIADB conditional is unnecessary, because we control how this content is used by inserting the include directive only in `mariadb.adoc`.
Because the heading is only a navigational aid within the partials file, and is not for use in the parent file, it should precede the tag region delimiter.
Initially, the content is specified for community use only, so we precede it with the `community` conditional.

> From MariaDB 12.3, the binary log can be optionally stored in InnoDB-managed, page-structured files.

Reviewer: ```suggestion
In MariaDB versions 12.3 and later, you can configure the database to store the binary log in InnoDB-managed, page-structured files (`.ibb`), instead of traditional flat files. 
```

> When enabled, this feature eliminates the traditional two-phase commit (2PC) protocol between the

Reviewer: ```suggestion
If you configure the database to store the binary log in InnoDB files, it does not enforce the two-phase commit (2PC) protocol between the
```

> binary log and InnoDB, providing improved commit latency and simplified crash recovery.

Reviewer: ```suggestion
binary log and InnoDB, resulting in improved commit latency and simplified crash recovery.
```

> .Important considerations

Reviewer: ```suggestion
When the database uses InnoDB to store the binary log, the following considerations apply: 
```
In lieu of the section title, add an introductory sentence to introduce the unordered list. Although this content is currently targeted for the community edition, if we later incorporate the content into the product doc in a future release, the conversion process gets complicated when subsections are present.

* Replication must use GTID-based positioning. Filename and offset-based positions are not available.

> * This feature is mutually exclusive with the traditional file-based binlog. The server writes only the InnoDB-based format.

Reviewer: ```suggestion
* Replication must use GTID-based positioning.
Filename and offset-based positions are not available.
* This feature is mutually exclusive with the traditional file-based binlog. 
The server uses the InnoDB-based format to store all entries.
```
Per Debezium contribution guidelines, enter a single sentence per line to improve readability of the source and simplify diffs for potential future updates.

Because the prerequisites list specifies that GTID replication is enabled, can we remove the first bullet here?

> * Galera Cluster is not supported when InnoDB-based binary logs are in use.

Reviewer: ```suggestion
* Galera Cluster is not supported.
```

* Semi-synchronous replication is not supported.
* The `sync_binlog` option is no longer required and is effectively ignored.

> * The {prodname} {connector-name} connector supports this feature from version {debezium-version}.

Reviewer: ```suggestion
```
Remove this sentence. The mention of the feature implies that it's supported (in the community edition). Also, because we'll increment the value of the attribute in subsequent releases, the statement has a short shelf life.

.Prerequisites

* MariaDB version 12.3 or later.
* GTID-based replication is enabled.

.Procedure

. Add the following options to your MariaDB configuration file:
+
[source,properties]
----
log_bin
binlog_storage_engine=innodb
binlog_format=ROW
binlog_row_image=FULL
----
+
The `log_bin` option must be specified without a value. The `binlog_storage_engine=innodb`

> setting enables the InnoDB-based binary log.

Reviewer: ```suggestion
The `log_bin` property must be present in the configuration, but it cannot contain a value. 
The `binlog_storage_engine=innodb` setting enables the database to use InnoDB to store the binary log.
```

> . Verify the binlog format by connecting to the server and running:

Reviewer: ```suggestion
. Verify the binlog format by connecting to the server and running the following SQL command:
```

+
[source,sql]
----
SHOW BINARY LOGS;
----
+
Files are named with an `.ibb` extension (for example, `binlog-000000.ibb`).

For additional information about the InnoDB-based binary log, see the

> link:https://mariadb.com/docs/server/server-management/server-monitoring-logs/binary-log/innodb-based-binary-log/[MariaDB documentation].

Reviewer: ```suggestion
For more information about the InnoDB-based binary log, see the https://mariadb.com/docs/server/server-management/server-monitoring-logs/binary-log/innodb-based-binary-log/[MariaDB documentation].
```

> endif::MARIADB[]

Reviewer: ```suggestion
endif::community[]
end::enabling-innodb-binlog[]
```
Remove the closing `MARIADB` conditional and insert the closing `community` delimiter, followed by the closing tag delimiter.

end::enabling-binlog[]

=== Enabling the InnoDB-based binary log (MariaDB 12.3 and later)

tag::enabling-innodb-binlog[]
ifdef::community[]
In MariaDB versions 12.3 and later, you can configure the database to store the binary log in InnoDB-managed, page-structured files (`.ibb`), instead of traditional flat files.
If you configure the database to store the binary log in InnoDB files, it does not enforce the two-phase commit (2PC) protocol between the

> binary log and InnoDB, resulting in improved commit latency and simplified crash recovery.

Reviewer 2: ```suggestion
If you configure the database to store the binary log in InnoDB files, it does not enforce the two-phase commit (2PC) protocol between the binary log and InnoDB, resulting in improved commit latency and simplified crash recovery.
```
Documentation should always be 1 sentence per line.

|`log_bin`
|The value of `log_bin` is the base name of the sequence of binlog files.
ifdef::MARIADB[]
When using the link:https://mariadb.com/docs/server/server-management/server-monitoring-logs/binary-log/innodb-based-binary-log/[InnoDB-based binary log]

> on MariaDB 12.3+, this option must be specified without a value.

Reviewer: ```suggestion
If the MariaDB server uses link:https://mariadb.com/docs/server/server-management/server-monitoring-logs/binary-log/innodb-based-binary-log/[InnoDB to store the binary log], add an empty `log_bin` property to the MariaDB configuration.
That is, list the property in the configuration, but do not specify a value.
```

endif::MARIADB[]

ifdef::MARIADB[]
|`binlog_storage_engine`
|Controls which storage engine manages the binary log. Set to `innodb` to enable the
link:https://mariadb.com/docs/server/server-management/server-monitoring-logs/binary-log/innodb-based-binary-log/[InnoDB-based binary log]
feature in MariaDB 12.3 and later. When set to `innodb`, the `log_bin` option must be specified

> without a value, and replication must use GTID-based positioning.

Reviewer: ```suggestion
|Specifies the storage engine that manages the binary log. 
Set the value to `innodb` to enable the https://mariadb.com/docs/server/server-management/server-monitoring-logs/binary-log/innodb-based-binary-log/[InnoDB-based binary log]
feature in MariaDB 12.3 and later. 
When set to `innodb`, the `log_bin` option must be specified with no value, and replication must use GTID-based positioning.
```

> If unset, the default value is `file`, which uses the traditional file-based binary log.

Reviewer: ```suggestion
If you do not set this property, the database uses the default value, `file`, and stores log data in standard flat file format.
```

|`log_bin`
|The value of `log_bin` is the base name of the sequence of binlog files.

> ifdef::MARIADB[]

Reviewer: ```suggestion
ifdef::MARIADB[]
ifdef::community[]
```

When using the link:https://mariadb.com/docs/server/server-management/server-monitoring-logs/binary-log/innodb-based-binary-log/[InnoDB-based binary log]
on MariaDB 12.3+, this option must be specified without a value.

> endif::MARIADB[]

Reviewer: ```suggestion
endif::community[]
endif::MARIADB[]
```

> ifdef::MARIADB[]

Reviewer: ```suggestion
ifdef::MARIADB[]
ifdef::community[]
```

|`binlog_storage_engine`
|Specifies the storage engine that manages the binary log.
Set the value to `innodb` to enable the https://mariadb.com/docs/server/server-management/server-monitoring-logs/binary-log/innodb-based-binary-log/[InnoDB-based binary log]
feature in MariaDB 12.3 and later.
When set to `innodb`, the `log_bin` option must be specified with no value, and replication must use GTID-based positioning.
If you do not set this property, the database uses the default value, `file`, and stores log data in standard flat file format.

> endif::MARIADB[]

Reviewer: ```suggestion
endif::community[]
endif::MARIADB[]
```

|`log_bin`
|The value of `log_bin` is the base name of the sequence of binlog files.
ifdef::MARIADB[]
ifdef::community[]

> When using the link:https://mariadb.com/docs/server/server-management/server-monitoring-logs/binary-log/innodb-based-binary-log/[InnoDB-based binary log]

Reviewer 2: I believe this fragment should have been removed, IIUC.
